block_my_feedback.php

```diff
@@ -59,15 +65,180 @@ public function get_content(): stdClass {
+        return null;
+    }
+
+    /**
+     * Return mod data - due date & require marking.
+     *
+     * TODO - turnitin, quiz.
+     *
+     * @param stdClass $mod
+     * @param stdClass $course
+     */
+    public function get_mod_data($mod, $assess) {
+        // Mods have different feilds for due date, and require marking.
+        switch ($mod->modname) {
+            case 'assign':
+
+                // Check mod due date is relevant.
+                $duedate = self::duedate_in_range($mod->customdata['duedate']);
+                if (!$duedate) {
+                    return false;
+                }
+
+                // Add dates.
+                $assess->unixtimestamp = $duedate;
+                $assess->duedate = date('jS F y', $duedate);
+
+                // Require marking.
+                $context = context_module::instance($mod->id);
+                $assignment = new assign($context, $mod, $mod->course);
+                $assess->requiremarking = $assignment->count_submissions_need_grading();
+                if (!$assess->requiremarking) {
+                    return false;
+                }
+                $assess->markingurl = new moodle_url('/mod/'. $mod->modname. '/view.php', ['id' => $assess->cmid, 'action' => 'grader']);
+                // Return template data.
+                return $assess;
+
+            // TODO - quiz
+            case 'quiz':
+                // return 'timeclose';
+                return false;
+            // TODO - turnitin
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * Return if course has started (startdate) and has not ended (enddate).
+     *
+     * @param stdClass $course
+     */
+    public function is_course_current(stdClass $course): bool {
+        // Start date.
+        if ($course->startdate > time()){
+            return false; // Before the start date.
+        }
+
+        // End date.
+        if (isset($course->enddate)) {
+            if ($course->enddate == 0) {
+                return true; // Enddate is set to 0 when no end date, show course.
+            }
+            if (time() > $course->enddate){
+                return false; // After the end date.
+            }
         }
+        return true; // All good, show course.
+    }
 
-        $this->content->text = $OUTPUT->render_from_template('block_my_feedback/content', $template);
-        return $this->content;
+    /**
+     * Return if a due date in the date range.
+     *
+     * @param int $duedate
+     */
+    public function duedate_in_range(int $duedate): ?int {
+        // Only show dates within a month.
+        $past = strtotime('-1 month');
+        $future = strtotime('+1 month');
+        // If due date is too far in the future.
+        if ($duedate > $future) {
+            return false;
+        }
+        // If due date is too far in the past.
+        if ($duedate < $past) {
+            return false;
+        }
+        return $duedate;
```

Comment:
Can't we make this just:

```
        return ($duedate > strtotime('-1 month')) && ($duedate < strtotime('+1 month'));
```

Reply:
Again, i'm keeping this verbose for legibility (plus fully expecting changes to this).